test_conformance/basic/test_int2fp.cpp

```diff
@@ -0,0 +1,316 @@
+//
+// Copyright (c) 2023 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "CL/cl_half.h"
+#include "harness/compat.h"
+#include "harness/errorHelpers.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#include <algorithm>
+#include <map>
+#include <vector>
+
+#include "procs.h"
+#include "utils.h"
+
+namespace {
+const char *int2float_kernel_code = R"(
+%s
+__kernel void test_X2Y(__global TYPE_X *src, __global TYPE_Y *dst)
+{
+    int  tid = get_global_id(0);
+
+    dst[tid] = (TYPE_Y)src[tid];
+
+})";
+
+template <bool int2fp> struct TypesIterator
+{
+    TypesIterator(cl_device_id deviceID, cl_context context,
+                  cl_command_queue queue, int num_elems, const char *test_name)
+        : context(context), queue(queue), test_name(test_name),
+          num_elements(num_elems)
+    {
+        fp16Support = is_extension_available(deviceID, "cl_khr_fp16");
+        fp64Support = is_extension_available(deviceID, "cl_khr_fp64");
+
+        type2name[sizeof(cl_half)] = std::make_pair("half", "short");
+        type2name[sizeof(cl_float)] = std::make_pair("float", "int");
+        type2name[sizeof(cl_double)] = std::make_pair("double", "long");
+
+        std::tuple<cl_float, cl_half, cl_double> it;
+        for_each_elem(it);
+    }
+
+    template <typename T> void generate_random_inputs(std::vector<T> &v)
+    {
+        RandomSeed seed(gRandomSeed);
+
+        if (sizeof(T) == sizeof(cl_half))
+        {
+            if (int2fp)
+            {
+                auto random_generator = [&seed]() {
+                    return (short)get_random_float(
```

Comment:
preference to use `cl_short` for cast rather than `short`

Reply:
Done